frontend/src/components/QuerySentencePreview/utils/index.ts

```diff
@@ -100,16 +100,26 @@ export const getClauseOptions = (
 export const resetClauseOptions = (
   options: AdvancedQueryOptions,
   clause?: AdvancedQueryBuilderAction,
+  clearAll?: boolean,
 ): AdvancedQueryOptions => {
   const _options = { ...options };
-  if (clause === 'select') {
+  if (clearAll) {
     delete _options.columns;
+    delete _options.filter;
+    delete _options.join;
+    delete _options.groupby;
```

Comment:
Remove these and simply return the default config... it's just the `source` & `selectall` - one operation instead of 5.

Reply:
Updated